### PR TITLE
Add docs for Callback config

### PR DESCRIPTION
**Pull request by kjbracey**
Documentation for changes in https://github.com/ARMmbed/mbed-os/pull/12036

### CODE REVIEW

**Comment by 40Grit** [comment]
@aglass0fmilk

**Comment by AnotherButler** [comment]
@iriark01 This is waiting on engineering review before editing.

**Comment by kjbracey** [comment]
Part of the change was moved to https://github.com/ARMmbed/mbed-os/pull/12761. These docs are largely fine if that doesn't make it, but the wording does strongly suggest that the default for the `callback-nontrivial` is `false`. It's `true` until 12761 lands.

**Comment by iriark01** [comment]
@kjbracey-arm was the intention to hold on this PR until 12761 makes it in?

**Comment by kjbracey** [comment]
I'm fine with it going now, but maybe it could be edited. That first paragraph talking about how the "setting can almost always be left set to false" implies that it is false by default, but it isn't yet.

Maybe just take out the word "left"? Then it works either way, I think.

File: docs/api/platform/Callback.md [hunk 1]
```diff
@@ -26,6 +26,16 @@ void run_timer_event(Callback<void(float)> on_timer) {
 
 The Callback class is what’s known in C++ as a “Concrete Type”. That is, the Callback class is lightweight enough to be passed around like an int, pointer or other primitive type.
 
+## Configuration
+
+Two system configuration options permit trade-offs between image size and flexibility of the Callback class.
+
+* `platform.callback-nontrivial` controls whether Callbacks can store non-trivially-copyable function objects. Having this setting off saves significant code size, as it makes Callback itself trivially-copyable, so all Callback assignments and copies are simpler. Almost all users use Callback only with function pointers, member function pointers or lambdas with trivial captures, so this setting can almost always be left set to false. A compile-time error will indicate that this setting needs to be set to true if any code attempts to assign a non-trivially-copyable object to a Callback.
```
Comment (iriark01) [inline docs/api/platform/Callback.md]:
```suggestion
* `platform.callback-nontrivial` controls whether Callbacks can store non-trivially-copyable function objects. Having this setting off saves significant code size, as it makes Callback itself trivially-copyable, so all Callback assignments and copies are simpler. Almost all users use Callback only with function pointers, member function pointers or lambdas with trivial captures, so this setting can almost always be set to false. A compile-time error will indicate that this setting needs to be set to true if any code attempts to assign a non-trivially-copyable object to a Callback.
```

**Comment by iriark01** [comment]
@kjbracey-arm looks like I can merge this?

**Comment by kjbracey** [comment]
Yes, thanks!

**Pull request merged**